### DISCUSSION TITLE
Add elements for customizing the login page

### DISCUSSION THREAD

**Discussion by Hir0-84** [dani-garcia/vaultwarden#3569] [Ideas]
Dear developers,

Would be worth to add the option to customize a bit the login page, like a custom background maybe with a transparent effect to avoid opening up for text color full customization?

I explain the reason of this request. At the very beginning, when I started using Vaultwarden I have chosen as my own local server domain name something identical to a public domain name, without knowing that it was already a registered name. My web request was forwarded outside my local network and I was actually trying to log in into a vaultwarden of someone else. 

That page could have been a fake one and I started to be a bit paranoic. So I decided to customize manually my web page editing the web-vault folder and without having any experience with HTLM I managed to add a background image, adapt the color of some text and my initial to the "Vaultwarden (unofficial Bitwarden® server)" message at the bottom of login page. 

Everytime I update my docker I risk to lose all those modifications but I am wondering if that will be useful to other people as well and add a bit of extra security. I am not asking to implement a full customizable web interface. Already uploading a different image as background or the possibility to insert a custom welcome message would help a lot.

What do you think? Thank you in advance for your time and thanks for the amazing work you have done

**Comment by BlackDex** [comment]
There isn't an easy way to do this, all is done in JavaScript or CSS and intercepting the call to the correct vault files and adjust them could also produce errors. It's not that we can easily just add some variables or something, since nothing of the web-vault is pre-processed.

We want to keep the web-vault as close as possible to the upstream.

Creating adjustments during startup manually is i think the most easy way. An other option would be using a reverse proxy which is capable of doing something like that, like nginx for example.
